test/WasmTasksTests/WasmTask_Tests.cs

```diff
@@ -160,42 +161,220 @@ private static void ExecuteCommand(string command)
-        public ITaskItem? OutputFile { get; set; }
+        public class ConcatWasmTask : WasmTask
+        {
+            public ITaskItem? InputFile1 { get; set; }
+            public ITaskItem? InputFile2 { get; set; }
+            [Output]
+            public ITaskItem? OutputFile { get; set; }
 
-        public ConcatWasmTask() : base()
+            public ConcatWasmTask() : base()
+            {
+                WasmFilePath = WasmTask_Tests.s_concatFilePath;
+                BuildEngine = new MockEngine();
+            }
+        }
+
+        public class DirectoryMergeWasmTask : WasmTask
         {
-            WasmFilePath = WasmTask_Tests.s_concatFilePath;
-            BuildEngine = new MockEngine();
+            public ITaskItem[]? Dirs { get; set; }
+            public string? MergedName { get; set; }
+            [Output]
+            public ITaskItem? MergedDir { get; set; }
+
+            public DirectoryMergeWasmTask() : base()
+            {
+                WasmFilePath = WasmTask_Tests.s_mergeFilePath;
+                BuildEngine = new MockEngine();
+            }
         }
-    }
+        private class CustomRustE2ETest
+        {
+            private readonly string _name;
+            private readonly string _templatePath;
+            private readonly string _testProjectPath;
 
-    public class DirectoryMergeWasmTask : WasmTask
-    {
-        public ITaskItem[]? Dirs { get; set; }
-        public string? MergedName { get; set; }
-        [Output]
-        public ITaskItem? MergedDir { get; set; }
+            public CustomRustE2ETest(string name, string toplevel, string executeRustCode, string getTaskInfoRustCode)
+            {
+                _name = name;
+                _templatePath = Path.Combine(SOLUTION_ROOT_PATH, s_paths[0]);
+                _testProjectPath = Path.Combine(TEST_PROJECT_PATH, "GeneratedRustTests", _name);
+
+                CopyTemplate();
+                CreateLibRs(toplevel, executeRustCode, getTaskInfoRustCode);
+                Compile();
+            }
+
+            private void CopyTemplate()
+            {
+                if (Directory.Exists(_testProjectPath))
+                {
+                    Directory.Delete(_testProjectPath, true);
+                }
+                DirectoryCopy(_templatePath, _testProjectPath, true);
+            }
+
+            private void CreateLibRs(string topLevel, string execute, string getTaskInfo)
+            {
+                string contents = $@"
+mod msbuild;
+use msbuild::logging::{{log_warning, log_error, log_message}};
+use msbuild::task_info::{{task_info, Property, PropertyType, TaskInfoStruct, TaskResult}};
+use serde::{{Serialize, Deserialize}};
+{topLevel}
+
+#[no_mangle]
+#[allow(non_snake_case)]
+pub fn Execute() -> TaskResult
+{{
+{execute}
+}}
+
+#[no_mangle]
+#[allow(non_snake_case)]
+pub fn GetTaskInfo() 
+{{
+{getTaskInfo}
+}}
+";
+                File.WriteAllText(Path.Combine(_testProjectPath, "src", "lib.rs"), contents);
```

Comment:
It can be handy for your tests
https://github.com/dotnet/msbuild/blob/0716e1c2aec9da7d6cd3c1df4707b0f1b6137868/src/UnitTests.Shared/TestEnvironment.cs#L677